private/pkg/filelock/locker.go

```diff
@@ -56,10 +72,17 @@ func (l *locker) RLock(ctx context.Context, path string, options ...LockOption)
 	if err := validatePath(path); err != nil {
 		return nil, err
 	}
+	options = append(
+		[]LockOption{
+			LockWithTimeout(l.lockTimeout),
+			LockWithRetryDelay(l.lockRetryDelay),
+		},
+		options..., // Any additional options set will be applied last
+	)
 	return rlock(
 		ctx,
 		normalpath.Unnormalize(normalpath.Join(l.rootDirPath, path)),
-		options...,
+		options..., // Any additional options set will be applied last
```

Comment:
```suggestion
		options...,
```

nit: Remove duplicate comment - makes more sense above on the append call.